website/pages/docs/secrets/databases/custom.mdx

```diff
@@ -12,81 +12,164 @@ description: |-
+commands to create the user (and often set permissions for that user). These statements are
+from the following fields in the API:
+
+| API Argument               | Request Object                                     |
+| -------------------------- | -------------------------------------------------- |
+| `creation_statements`      | `NewUserRequest.Statements.Commands`               |
+| `revocation_statements`    | `DeleteUserRequest.Statements.Commands`            |
+| `rollback_statements`      | `NewUserRequest.RollbackStatements.Commands`       |
+| `renew_statements`         | `UpdateUserRequest.Expiration.Statements.Commands` |
+| `rotation_statements`      | `UpdateUserRequest.Password.Statements.Commands`   |
+| `root_rotation_statements` | `UpdateUserRequest.Password.Statements.Commands`   |
+
+In many of the built in plugins, they replace `{{name}}` (or `{{username`), `{{password}}`,
+and/or `{{expiration}}` with the associated values. It is up to your plugin to perform these
+string replacements. There is a helper function located in `sdk/database/helper/dbutil`
+called `QueryHelper` that assists in doing this string replacement. You are not required to
+use it, but it will make your plugin's behavior consistent with the built in plugins.
+
+The `InitializeRequest` object contains a map of keys to values. This data is what the
 user specified as the configuration for the plugin. Your plugin should use this
-data to make connections to the database. It is also passed a boolean value
-specifying whether or not your plugin should return an error if it is unable to
-connect to the database.
+data to make connections to the database. The response object contains a similar configuration
+map. The response object should contain the configuration map that should be saved within Vault.
+This allows the plugin to manipulate the configuration prior to saving it.
+
+It is also passed a boolean value indicating if your plugin should initialize a connection to
+the database during the `Initialize` call. This function is called when the configuration is
+written. This allows the user to know whether the configuration is valid and able to connect
+to the database in question. If this is set to false, no connection should be made during the
+`Initialize` call, but subsequent calls to the other functions will need to open a connection.
 
 ## Serving your plugin
 
-Once your plugin is built you should pass it to vault's `plugins` package by
-calling the `Serve` method:
+The plugin runs as a separate binary outside of Vault, so the plugin itself will need a `main`
+function. Use the `Serve` function within `sdk/database/dbplugin/v5` to serve your plugin. You
+will also need to pass some TLS configuration information that Vault uses when initializing the
+plugin. Below is an example setup:
 
 ```go
 package main
 
 import (
-    "github.com/hashicorp/vault/api/plugins"
+	"github.com/hashicorp/vault/api/plugins"
```

Comment:
Not sure if intentional, but spacing is set to 1 space on per tab in here.

Reply:
<img src=https://media.giphy.com/media/qxtxlL4sFFle/giphy.gif height=150>

That's a tab in every other view except github's diff view 0.o